pkg/daemon/ceph/client/osd.go

```diff
@@ -220,6 +221,45 @@ func GetOSDUsage(context *clusterd.Context, clusterInfo *ClusterInfo) (*OSDUsage
 	return &osdUsage, nil
 }
 
+func convertKibibytesToTebibytes(kib string) (float64, error) {
+	kibFloat, err := strconv.ParseFloat(kib, 64)
+	if err != nil {
+		return float64(0), errors.Wrap(err, "failed to convert string to float")
+	}
+	return kibFloat / float64(1024*1024*1024), nil
+}
+
+func ResizeOsdCrushWeight(actualOSD OSDNodeUsage, ctx *clusterd.Context, clusterInfo *ClusterInfo) (bool, error) {
+	crushWeight, err := strconv.ParseFloat(actualOSD.CrushWeight.String(), 64)
+	if err != nil {
+		return false, errors.Wrapf(err, "failed converting string to float for osd.%d crush weight %q", actualOSD.ID, actualOSD.CrushWeight.String())
+	}
+	// actualOSD.KB is in KiB units
+	desiredOSDTiB, err := convertKibibytesToTebibytes(actualOSD.KB.String())
+	if err != nil {
+		return false, errors.Wrapf(err, "failed to convert KiB to TiB for osd.%d crush weight %q", actualOSD.ID, actualOSD.KB.String())
+	}
+
+	// do not reweight if the desired crush weight is 0 or they are equal or there percentage resize is less than 10 percent
+	if desiredOSDTiB == float64(0) {
+		logger.Debugf("desired osd size is 0 for osd.%d, not resizing the crush weights", actualOSD.ID)
+		return false, nil
+	} else if math.Abs(((desiredOSDTiB - crushWeight) / crushWeight)) <= 0.1 {
+		logger.Debugf("desried osd size %f is less then 10 percent increased from current osd size %f for osd.%d, not resizing the crush weights", desiredOSDTiB, crushWeight, actualOSD.ID)
```

Comment:
It seems like we also need to add a check to ensure the new weight is greater than the old weight. 

Would adding this make sense?
```go
if newWeight < oldWeight {
    logger.Infof("new weight less than existing weight, not updating etc....")
}
```